### PR TITLE
[cmd/configschema] Remove `cmd/configschema`

### LINKED ISSUE #30187: configschema future
### Component(s)

_No response_

### Describe the issue you're reporting

`configschema` imports all the components of the repository and needs upkeep to generate configuration. It is not used by OpenTelemetry currently and maintainers are not aware of its design.

This is similar to #27849 - we want to generate docs per module instead.

The experiment #30186 shows a possible path forward.

As result, we can use the following path forward:
* [ ] Duplicate the logic of `cmd/configschema` into `internal/configschema` so it becomes available as a library
       Offer this new approach via mdatagen, so it generates docs for components opting in
       Enable this for at least one component to show the effect
       #30188
* [x] #30190
* [ ] Delete configschema and `internal/components`

### PR DESCRIPTION
**Pull request by TylerHelmuth**
**Description:**
Removes the deprecated `cmd/configschema`.

The command is not used in our repository and is a source of dependency pain 

**Link to tracking Issue:**
Closes https://github.com/open-telemetry/opentelemetry-collector-contrib/issues/30187

### CODE REVIEW

File: .github/ALLOWLIST [hunk 1]
```diff
@@ -25,3 +25,4 @@ internal/common
 
 ## UNMAINTAINED components
 exporter/skywalkingexporter/
+receiver/googlecloudspannerreceiver/
```
Comment (crobert-1) [inline .github/ALLOWLIST:28]:
```suggestion
```
This looks unrelated to this PR, we should probably open a separate PR if necessary. Same goes for all other references to the `googlecloudspannerreceiver`

Comment (TylerHelmuth) [reply .github/ALLOWLIST:28]: https://github.com/open-telemetry/opentelemetry-collector-contrib/pull/33385

**Review by atoulme — APPROVED** [review]

**Review by crobert-1 — COMMENTED** [review]
Just searching some references here.

`Makefile`: The [`genconfigdocs` target](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/76dc438a6f728474791273a1cd71e4c45b71d355/Makefile#L482) should be removed.
[`.golangci.yml`](https://github.com/open-telemetry/opentelemetry-collector-contrib/blob/76dc438a6f728474791273a1cd71e4c45b71d355/.golangci.yml#L177): There's an exclude rule referencing this directory, the `staticcheck` linter is disabled due to this package. Should we remove the exclude rule?

**Comment by TylerHelmuth** [comment]
@crobert-1 nice finds

**Review by crobert-1 — APPROVED** [review]

**Review by MovieStoreGuy — APPROVED** [review]

**Review by songy23 — APPROVED** [review]

**Review by codeboten — APPROVED** [review]
